client/src/components/Navbar/Navbar.js

```diff
@@ -1,21 +1,73 @@
-import React from 'react';
+import React, { Component } from "react";
 import "./Navbar.css";
-import Logo from "../Logo/Logo"
+import Logo from "../Logo/Logo";
+import Burger from "../Burger/Burger";
+import ProfilePicture from "../ProfilePicture/ProfilePicture";
+import NavLink from "../NavLink/NavLink";
+import Icon from "../Icon/Icon";
+import PropTypes from "prop-types";
 
-// Function based React Component
-const Navbar = (props) => {
+// Class Based React Component
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+    // Default CSS class to apply to the Component
+    this.state = {
+      condensedMenuActive: false,
+      // menuItems are careated in order of the following array.
+      menuItems: ["My Directory", "Community", "Forum", "Book a Trip"],
+    };
+  }
 
-  // Default Class to apply to Component
-  let classList = `Navbar`;
+  clickHandler = () => {
+    console.log("Clicked!")
+  }
 
-
-  return (
-    <div className={classList}>
-      <div className="navbar-logo">
-        <Logo clickable to="/route" />
+  render() {
+    return (
+      <div className="Navbar">
+        <div className="Navbar__logo">
+          <Logo clickable to="/" />
+        </div>
+        <div className="Navbar__burger">
+          {/* Handles activating the hamburger animation and displays the menu. */}
+          <Burger
+            onClick={() => {
+              this.setState({ condensedMenuActive: !this.state.condensedMenuActive });
+            }}
+            active={this.state.condensedMenuActive}
+          />
+        </div>
+        <div className={`Navbar__main ${this.state.condensedMenuActive && `Navbar__show`}`}>
+          <div className="Navbar__menu">
+            {this.state.menuItems.map((item, key) => (
+              <div className="Navbar__link-item" key={key}>
+                <NavLink text={item} to={"/"} menuList={[]} />
+                {this.props.page === key && (
+                  <div className={`Navbar__triangle`}></div>
+                )}
+              </div>
+            ))}
+          </div>
+          <div className="Navbar__right">
+            <Icon icon={"search"} size="2x" onClick={this.clickHandler} />
+            <Icon icon={["far", "comment-dots"]} size="2x" onClick={this.clickHandler} />
+            <ProfilePicture type="extra-small" to="/" image={this.props.profileImage} />
```

Comment:
Just caught this - this should link to the profile page for the current user. Could we have this be a prop that gets passed down?

Reply:
Done. Added new prop to be passed called profileUrl.